File: CSharp/One/Transforms/ConvertToMethodCall.cs
using One.Ast;
using One;

namespace One.Transforms
{
    public class ConvertToMethodCall : AstTransformer {
        public ConvertToMethodCall(): base("ConvertToMethodCall")
        {
            
        }
        
        protected override Expression visitExpression(Expression expr)
        {
            var origExpr = expr;
            
            expr = base.visitExpression(expr);
            
            if (expr is BinaryExpression binExpr && binExpr.operator_ == "in")
                expr = new UnresolvedCallExpression(new PropertyAccessExpression(binExpr.right, "hasKey"), new IType[0], new Expression[] { binExpr.left });
            
            expr.parentNode = origExpr.parentNode;
            return expr;
        }
    }
}

File: PHP/One/Transforms/ConvertToMethodCall.php
<?php

namespace One\Transforms\ConvertToMethodCall;

use One\Ast\Expressions\Expression;
use One\Ast\Expressions\ElementAccessExpression;
use One\Ast\Expressions\UnresolvedCallExpression;
use One\Ast\Expressions\PropertyAccessExpression;
use One\Ast\Expressions\BinaryExpression;
use One\Ast\Expressions\StringLiteral;
use One\AstTransformer\AstTransformer;

class ConvertToMethodCall extends AstTransformer {
    function __construct() {
        parent::__construct("ConvertToMethodCall");
        
    }
    
    protected function visitExpression($expr) {
        $origExpr = $expr;
        
        $expr = parent::visitExpression($expr);
        
        if ($expr instanceof BinaryExpression && $expr->operator === "in")
            $expr = new UnresolvedCallExpression(new PropertyAccessExpression($expr->right, "hasKey"), array(), array($expr->left));
        
        $expr->parentNode = $origExpr->parentNode;
        return $expr;
    }
}
